contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,541 @@
+    Motion storage motion = motions[_motionId];
+
+    // TODO: Check per-permission
+    bytes32 motionPermissions = motion.requiredPermissions;
+    uint8 roleIndex = 0;
+    while (motionPermissions > 0){
+      if (uint256(motionPermissions)%2 == 1){
+        uint256 threshold = getDomainSkillRoleThreshold(motion.domainSkillId, ColonyRole(roleIndex));
+        require(motionRoleApprovalCount[_motionId][ColonyRole(roleIndex)] >= threshold, "colony-multisig-permissions-not-enough-approvals");
+      }
+      motionPermissions = motionPermissions >> 1;
+      roleIndex += 1;
+    }
+
+    // If approvals were made, threshold lowered, and then executed, motion.overallApprovalTimestamp is 0
+    if (motion.overallApprovalTimestamp == 0){
+      // We set the overall approval timestamp to now, and return
+      // We don't execute the motion, but we want to commit the timestamp
+      // (which wouldn't happen if we continued and the call(s) failed)
+      motion.overallApprovalTimestamp = block.timestamp;
+      return;
+    }
+
+    bool overallSuccess = true;
+    motion.executed = true;
+
+    // Execute the motion, if it meets the threshold
+    for (uint256 i = 0; i < motion.data.length; i += 1){
+      (bool success, bytes memory result) = address(motion.targets[i]).call(motion.data[i]);
+      overallSuccess = overallSuccess && success;
+      // Allow failing execution after seven days
+      require(success || motion.overallApprovalTimestamp + 7 days <= block.timestamp, "colony-multisig-failed-not-one-week");
+    }
+
+    emit MotionExecuted(msgSender(), _motionId, overallSuccess);
+  }
+
+
+  function getMotionCount() public view returns(uint256) {
+    return motionCount;
+  }
+
+  function getMotion(uint256 motionId) public view returns (Motion memory){
+    return motions[motionId];
+  }
+  function getMotionRoleApprovalCount(uint256 _motionId, ColonyRole _role) public view returns (uint256){
+    return motionRoleApprovalCount[_motionId][_role];
+  }
+
+  function getUserApproval(uint256 _motionId, address _user, ColonyRole _role) public view returns (bool){
+    return motionApprovals[_motionId][_user][_role];
+  }
+
+  function getDomainSkillRoleCounts(uint256 _domainSkillId, uint8 role) public view returns (uint256) {
+    return domainSkillRoleCounts[_domainSkillId][role];
+  }
+
+  function getDomainSkillThreshold(uint256 _domainSkillId) public view returns (uint256) {
+    return domainSkillThreshold[_domainSkillId];
+  }
+
+  function setDomainSkillThreshold(uint256 _domainSkillId, uint256 _threshold) public returns (uint256) {
+    require(colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root), "multisig-permissions-not-core-root");
+    domainSkillThreshold[_domainSkillId] = _threshold;
+  }
+
+  function getDomainSkillRoleThreshold(uint256 _domainSkillId, ColonyRole _role) public view returns (uint256) {
+    // If domain skill threshold explicitly set, use that value
```

Comment:
Is there a reason this isn't structured as an if/else?

Reply:
Only the [training I've had from ESLint](https://eslint.org/docs/latest/rules/no-else-return).